test/blueprint/keyframe.spec.ts

```diff
@@ -0,0 +1,479 @@
+		],
+		data: {
+			// TODO: Should this be embedded or a link to config manifest file?
+			// Example: https://github.com/product-os/jellyfish/blob/master/deploy-templates/jellyfish-product/product/config-manifest.yml
+			'behaviour-config': [
+				{ name: 'NODE_ENV', type: 'string', secret: false },
+				{ name: 'LOGLEVEL', type: 'string', secret: false },
+			],
+		},
+		provides: [{ type: 'endpoint', as: 'api-endpoint', data: { port: 8000 } }],
+	};
+
+	const jfActionServerContract = {
+		type: 'sw.containerized-worker',
+		slug: 'jellyfish-action-server',
+		requires: [
+			// Note: external services are not covered atm. Configured as behaviour for now.
+			{ type: 'sw.database', slug: 'postgres-db' },
+			{ type: 'sw.containerized-service', slug: 'redis' },
+		],
+	};
+
+	const jfTickServerContract = {
+		type: 'sw.containerized-worker',
+		slug: 'jellyfish-tick-server',
+		requires: [
+			{ type: 'sw.database', slug: 'postgres-db' },
+			{ type: 'sw.containerized-service', slug: 'redis' },
+		],
+	};
+
+	const jfLivechatContract = {
+		type: 'sw.containerized-service',
+		slug: 'jellyfish-livechat',
+		requires: [{ type: 'sw.containerized-service', slug: 'jellyfish-api' }],
+	};
+
+	// Generic components.
+
+	const redisContract = {
+		type: 'sw.database',
+		slug: 'redis',
+		requires: [],
+	};
+
+	const postgresDbContract = {
+		type: 'sw.database',
+		slug: 'postgres-db',
+		requires: [],
+	};
+
+	// Classes.
+
+	const endpoint = {
+		type: 'interface',
+		slug: 'endpoint',
+		provides: [{ type: 'string', as: 'connection-uri' }],
+	};
+
+	const containerizedService = {
+		type: 'component-class',
+		slug: 'sw.containerized-service',
+		provides: [{ type: 'endpoint', as: 'default-endpoint' }],
+	};
+
+	const containerizedWorker = {
+		type: 'component-class',
+		slug: 'sw.containerized-worker',
+		// Main difference from the service: it provides nothing.
+		provides: [],
+	};
+
+	const database = {
+		type: 'component-class',
+		// Equivalent to the service for now. But can be useful to keep it separate.
+		// For instance, to underline it persists the state.
+		slug: 'sw.database',
+		provides: [{ type: 'endpoint', as: 'default-endpoint' }],
+	};
+
+	// The universe (we are missing versions for now).
+	const universe = [
+		jfApiContract,
+		jfActionServerContract,
+		jfTickServerContract,
+		jfUiContract,
+		jfLivechatContract,
+		postgresDbContract,
+		redisContract,
+	];
+
+	// productOS blueprint.
+
+	// Blueprint format is changed comparing to existing examples.
+	const productOsBlueprint = {
+		type: 'blueprint',
+		slug: 'productOs',
+
+		data: {
+			// Selector now queries contracts giving an instance name ("as"). Slug is used by default.
+			selector: [
+				{ slug: 'jellyfish-api', as: 'api' },
+				{ slug: 'jellyfish-ui', as: 'ui' },
+				{ slug: 'jellyfish-action-server', as: 'action-server' },
+				{ slug: 'jellyfish-tick-server', as: 'tick-server' },
+				{ slug: 'jellyfish-livechat', as: 'livechat' },
```

Comment:
This is a suggestion to unify `requires` in contracts and `selector` in blueprints, also adding an ability to refer to a specific instance with a custom name.
Compare to the selector here:
https://github.com/balena-io/contracts/blob/11cf69c31e6895b0c5bd88ece72ca1508183125b/contracts/sw.blueprint/device-instructions/contract.json

Why will not this work?